### PR TITLE
feat: view client charge overview 

### PR DESCRIPTION
**Pull request by BLasan**
Description
Component to view client charges completed.

Related issues and discussion
#680 

Screenshots, if any
webapp1

Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by BLasan** [comment]
@abhaychawla  Please review the PR

File: src/app/clients/clients-view/charges-overview/charges-overview.component.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { ActivatedRoute} from '@angular/router';
+
+@Component({
+  selector: 'mifosx-charges-overview',
+  templateUrl: './charges-overview.component.html',
+  styleUrls: ['./charges-overview.component.scss']
+})
+export class ChargesOverviewComponent implements OnInit {
+
+  displayedColumns: string[] = ['name', 'due_as_of', 'due', 'paid', 'waived', 'outstanding'];
```
Comment (karantakalkar) [inline src/app/clients/clients-view/charges-overview/charges-overview.component.ts]: use camel casing only

File: src/app/clients/clients-view/charges-overview/charges-overview.component.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      
+      </div>
+
+    </mat-card-content>
+
+  </mat-card>
+
+</div>
+
+
```
Comment (karantakalkar) [inline src/app/clients/clients-view/charges-overview/charges-overview.component.html]: remove extra spaces

File: src/app/clients/clients-view/charges-overview/charges-overview.component.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    this.setLoanClientChargeOverview();
+  }
+
+  /** Set Client Charge Overview. */
+  setLoanClientChargeOverview() {
+    this.dataSource = new MatTableDataSource(this.chargeOverviewData.pageItems);
+    this.dataSource.paginator = this.paginator;
+  }
+
+
```
Comment (karantakalkar) [inline src/app/clients/clients-view/charges-overview/charges-overview.component.ts]:
```suggestion

```

File: src/app/clients/clients-view/general-tab/general-tab.component.ts [hunk 4]
```diff
@@ -27,6 +27,7 @@ export class GeneralTabComponent implements OnInit {
   showClosedLoanAccounts = false;
   showClosedSavingAccounts = false;
   showClosedShareAccounts = false;
+  routeLink: any;
```
Comment (karantakalkar) [inline src/app/clients/clients-view/general-tab/general-tab.component.ts]: describe this variable properly

Comment (BLasan) [reply src/app/clients/clients-view/general-tab/general-tab.component.ts]: Comment added

**Comment by OussEmaDevCode** [comment]
@BLasan change the commit message as I mentioned in your last PR

File: src/app/clients/clients-view/charges-overview/charges-overview.component.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import { ActivatedRoute} from '@angular/router';
+
+@Component({
+  selector: 'mifosx-charges-overview',
+  templateUrl: './charges-overview.component.html',
+  styleUrls: ['./charges-overview.component.scss']
+})
+export class ChargesOverviewComponent implements OnInit {
+
+  displayedColumns: string[] = ['name', 'dueAsOf', 'due', 'paid', 'waived', 'outstanding'];
```
Comment (OussEmaDevCode) [inline src/app/clients/clients-view/charges-overview/charges-overview.component.ts]:
```suggestion
  /** Columns to be displayed in charge overview table. */
  displayedColumns: string[] = ['name', 'dueAsOf', 'due', 'paid', 'waived', 'outstanding'];
```

File: src/app/clients/clients-view/charges-overview/charges-overview.component.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
+/** Angular Imports */
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatTableDataSource, MatPaginator, MatDialog } from '@angular/material';
+import { ActivatedRoute} from '@angular/router';
```
Comment (OussEmaDevCode) [inline src/app/clients/clients-view/charges-overview/charges-overview.component.ts]:
```suggestion
import { ActivatedRoute } from '@angular/router';
```

**Comment by BLasan** [comment]
Travis build fails due to time out

**Comment by karantakalkar** [comment]
Can you refactor this for consistency, this will be merged tommorow.

**Comment by BLasan** [comment]
> Can you refactor this for consistency (especially the view), this will be merged tomorrow.

Removing mat-title and the mat-divider ? Any refactors to be done to the table?

**Comment by karantakalkar** [comment]
remove white background behind the table, also see other mat table views in app.

**Comment by BLasan** [comment]
> remove white background behind the table, also see other mat table views in app.

I'll re-correct it soon

**Comment by BLasan** [comment]
@karantakalkar Please review this

File: src/app/clients/clients-routing.module.ts [hunk 7]
```diff
@@ -149,7 +151,15 @@ const routes: Routes = [
...
+  },
+  {
+    path: 'clients/:clientId/chargeoverview',
+    component: ChargesOverviewComponent,
+    data: { title: extract('Client Charge Overview'), breadcrumb: 'Client Charge Overview', routeParamBreadcrumb: false },
+    resolve: {
+      clientChargesData: ClientChargeOverviewResolver,
+    }
+  },
   ])
```
Comment (karantakalkar) [inline src/app/clients/clients-routing.module.ts:163]: routing is a problem here as well

File: src/app/clients/clients-view/charges-overview/charges-overview.component.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+   * Retrieves the charge overview data from `resolve`.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {MatDialog} dialog Dialog reference.
+   */
+  constructor(private route: ActivatedRoute,
+              public dialog: MatDialog) {
+      this.route.data.subscribe((data: { clientChargesData: any }) => {
+        this.chargeOverviewData = data.clientChargesData;
+      });
+     }
```
Comment (karantakalkar) [inline src/app/clients/clients-view/charges-overview/charges-overview.component.ts:36]: reduce indent of this brace

File: src/app/clients/clients-view/general-tab/general-tab.component.ts [hunk 9]
```diff
@@ -27,6 +27,8 @@ export class GeneralTabComponent implements OnInit {
   showClosedLoanAccounts = false;
   showClosedSavingAccounts = false;
   showClosedShareAccounts = false;
+  /** Route Link to Charges Overview. */
+  routeLink: any;
```
Comment (karantakalkar) [inline src/app/clients/clients-view/general-tab/general-tab.component.ts:31]: chargesOverviewRoute

Comment (karantakalkar) [inline src/app/clients/clients-view/general-tab/general-tab.component.ts:31]:
```suggestion
```

File: src/app/clients/clients-routing.module.ts (hunk 7 above)
Comment (karantakalkar) [reply src/app/clients/clients-routing.module.ts:163]: nest the clients and client id here itslef and add route param breadcrumb, breadcrumb should be `home / clients / 1 / charges overview

Comment (karantakalkar) [reply src/app/clients/clients-routing.module.ts:163]: make the same change for previous PR approved PR as well

Comment (BLasan) [reply src/app/clients/clients-routing.module.ts:163]: I'll refactor it

**Comment by karantakalkar** [comment]
@BLasan fix routing this similar to payment form

File: src/app/clients/clients-routing.module.ts (hunk 7 above)
Comment (BLasan) [reply src/app/clients/clients-routing.module.ts:169]: Nested routings added. Please review

**Comment by BLasan** [comment]
> @BLasan fix routing this similar to payment form

It's done. Please have a look

**Comment by BLasan** [comment]
> Thanks @BLasan, great work today!, Muskan and Abhay will have a look and we'll merge by tommorow.

Okay, thanks for the guidance to make a quality PR :+1:

**Comment by muskankhedia** [comment]
@BLasan you can implement the module in the client view page itself as finalized in PR #690. For the implementation of the routes, you can refer to the file `clients-routing.module.ts` in PR #626.

**Comment by BLasan** [comment]
@karantakalkar @muskankhedia Please review

File: src/app/clients/clients-view/charges-overview/charges-overview.component.html [hunk 10]
```diff
@@ -0,0 +1,46 @@
+<div class="tab-container mat-typography">
+
+  <div style="margin-top: 5px;">
+
```
Comment (muskankhedia) [inline src/app/clients/clients-view/charges-overview/charges-overview.component.html:4]: Please add a heading with `Charges Overview` in order the user should know the page it is displayed.

Comment (BLasan) [reply src/app/clients/clients-view/charges-overview/charges-overview.component.html:4]: Above the table?

Comment (muskankhedia) [reply src/app/clients/clients-view/charges-overview/charges-overview.component.html:4]: yes.

Comment (BLasan) [reply src/app/clients/clients-view/charges-overview/charges-overview.component.html:4]: Ok. But what's the point of having breadcrumbs then?

Comment (BLasan) [reply src/app/clients/clients-view/charges-overview/charges-overview.component.html:4]: Done. Please review

**Pull request merged**